internal/controller/datadogagent/controller_reconcile_ccr.go

```diff
@@ -119,3 +121,40 @@ func deleteStatusWithClusterChecksRunner(newStatus *datadoghqv2alpha1.DatadogAge
 	newStatus.ClusterChecksRunner = nil
 	datadoghqv2alpha1.DeleteDatadogAgentStatusCondition(newStatus, datadoghqv2alpha1.ClusterChecksRunnerReconcileConditionType)
 }
+
+// cleanupOldCCRDeployments deletes CCR deployments when a CCR Deployment's name is changed using clusterChecksRunner name override
+func (r *Reconciler) cleanupOldCCRDeployments(ctx context.Context, logger logr.Logger, dda *datadoghqv2alpha1.DatadogAgent, newStatus *datadoghqv2alpha1.DatadogAgentStatus) error {
+	matchLabels := client.MatchingLabels{
+		apicommon.AgentDeploymentComponentLabelKey: datadoghqv2alpha1.DefaultClusterChecksRunnerResourceSuffix,
+		kubernetes.AppKubernetesManageByLabelKey:   "datadog-operator",
+	}
+	deploymentName := getDeploymentNameFromCCR(dda)
+	validDeploymentNames := map[string]struct{}{
```

Comment:
Since there will only be one CCR (and DCA) deployment in a cluster, I think it'll be simpler if we don't create a map for this. We can directly compare the valid deployment name against the `deployment.Name` in line 140